### PR TITLE
User Query Scope by permissions

### PR DESCRIPTION
**Pull request by iroes**
I suggest adding a query scope to get Users by a certain permission.

For example, to notify them depending on their capabilities:

```php
Notification::send(
    User::byAccess('platform.systems.users')->get(), 
    new NewUserVerifiedPendingReviewNotification(auth()->user())
);
```

### CODE REVIEW

**Comment by tabuna** [comment]
Could you please add tests?

**Comment by iroes** [comment]
I've added some tests.

**Comment by tabuna** [comment]
Can you remove the changes not related to the topic of permission checking?

**Comment by iroes** [comment]
Excuse me but which ones do you mean? The 'some php-cs-fixes' are all within the new code...

File: resources/views/layouts/pagination.blade.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ class="btn btn-sm btn-link dropdown-toggle p-0 m-0"
                         ->except(['page','_token']))
                         ->links('platform::partials.pagination')
                 !!}
-            @else
+            @elseif($paginator instanceof \Illuminate\Contracts\Pagination\Paginator)
```
Comment (tabuna) [inline resources/views/layouts/pagination.blade.php:39]: This is not related to permission checks

File: resources/views/layouts/table.blade.php [hunk 2]
```diff
@@ -63,7 +63,10 @@
         </div>
     @endif
 
-    @includeWhen(($rows instanceof \Illuminate\Contracts\Pagination\Paginator || $rows instanceof \Illuminate\Contracts\Pagination\CursorPaginator) && $rows->isNotEmpty(),
+    @includeWhen(
```
Comment (tabuna) [inline resources/views/layouts/table.blade.php:66]: This is not related to permission checks

File: resources/views/layouts/pagination.blade.php (hunk 1 above)
Comment (iroes) [reply resources/views/layouts/pagination.blade.php:39]: Oops sorry, I'll fix it as soon as I get home.

Comment (iroes) [reply resources/views/layouts/pagination.blade.php]: Done!
